### PR TITLE
Gh 974 fix logging treq

### PR DESCRIPTION
**Pull request by sam-falvo**

### CODE REVIEW

File: otter/util/logging_treq.py [hunk 1]
```diff
@@ -18,9 +19,14 @@ def _log_request(treq_call, url, **kwargs):
 
     :param callable f: a ``treq`` request method, such as ``treq.request``, or
         ``treq.get``, ``treq.post``, etc.
+    :param log: If provided, an instance of BoundLog.
+        Defaults to ``otter.log.log`` if not provided.
```
Comment (lvh) [inline otter/util/logging_treq.py]: Looks like it defaults to `otter.log.default_log`.

File: otter/auth.py [hunk 2]
```diff
@@ -61,6 +61,21 @@
 from otter.util.retry import repeating_interval, retry, retry_times
 
 
+class _DoNothingLogger(BoundLog):
```
Comment (cyli) [inline otter/auth.py]: I think we already have something like this - `otter.test.utils`:  https://github.com/rackerlabs/otter/blob/master/otter/test/utils.py#L299

Would that work for these purposes?  It does call mock, so you can assert stuff about logging if you wish, but it also will eat all logging messages and you do not have to check them.  :)

Comment (sam-falvo) [inline otter/auth.py]: I don't think that's a safe thing to do.  Production code should not depend on test code.

File: otter/util/logging_treq.py [hunk 3]
```diff
@@ -18,9 +18,14 @@ def _log_request(treq_call, url, **kwargs):
 
     :param callable f: a ``treq`` request method, such as ``treq.request``, or
         ``treq.get``, ``treq.post``, etc.
+    :param log: If provided, an instance of BoundLog.
```
Comment (cyli) [inline otter/util/logging_treq.py]: Yay, thank you for documenting this!

File: otter/auth.py (hunk 2 above)
Comment (cyli) [inline otter/auth.py]: True, thanks :)

**Comment by cyli** [comment]
LGTM!

**Pull request merged**